.github/workflows/test.yml

```diff
@@ -54,9 +54,8 @@ jobs:
     - name: Generate coverage report
       working-directory: ./python
       run: |
-        pip install -r requirements.txt
-        pip install coverage
+        python -I -m pip install 'coverage<8,>=7' pyspark==3.2.1 -r requirements.txt
```

Comment:
I think we need to update .github/workflows/onrelease.yml too

Reply:
Yes. Longer term, I think we should clean up & simplify the GH actions workflows (see issue #319), but for now, we'll need to update both of these if there are build changes.